src/components/slider/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import {
+  Slider as ChakraSlider,
+  SliderMark as ChakraSliderMark,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+} from '@chakra-ui/react';
+
+type SliderMark<T> = {
+  stepValue?: number;
+  label: string;
+  value: T;
+};
+
+type SliderProps<T> = {
+  useIndentation?: boolean;
+  marks: SliderMark<T>[];
+  defaultValue: SliderMark<T>;
+  onValueChanged: (arg: T) => void;
+};
+
+const getDefaultStepValue = <T,>(
+  marks: SliderMark<T>[],
+  step: number = 1,
+  useIndentation: boolean = false,
+  defaultValue: SliderMark<T>,
+) => {
+  return (
+    (defaultValue && (marks.indexOf(defaultValue) + +useIndentation) * step) ||
+    step - +!useIndentation
```

Comment:
I have mixed feelings about this `+!useIndentation`. It took me a while to figure out what is happining here. Can we simplify this a bit, maybe try not to use `+!useIndentation`. What do you think?